# Analizando dataset: Indicadores de Ovinos 

## Introducción

Este dataset se obtuvo de [Datos Argentina](https://www.datos.gob.ar/dataset/ssprys-indicadores-sectoriales-provinciales/archivo/ssprys_01.27). Se trata de un dataset muy poblado que contiene información sobre existencias ovinas, producción de lana y producción de carne tomadas de distintas fuentes como [MAGyP](https://www.argentina.gob.ar/agricultura/ganaderia), [FAO](https://www.fao.org/argentina/es/), [FLA](https://www.flasite.com/index.php/es/) y [SENASA](https://www.argentina.gob.ar/senasa).

Nos encontramos con indicadores anuales, trimestrales y mensuales. No existe documentación que nos proporcione información sobre el dataset, así que vamos a cargarlo para ver su contenido y comenzar el análisis.

## Análisis

### Importar librerias

Importamos las librerías de Python que necesitamos para analizar el dataset.

In [1]:
import pandas as pd

### Cargar dataset

Cargamos el dataset y observamos su contenido.

In [2]:
df = pd.read_csv('datos/ovinos-lana-y-carne.csv', sep=",", encoding='latin-1')
df

sector_id          sector_nombre  variable_id actividad_producto_nombre  \
0             1  Ovinos - Lana y Carne            1               Carne ovina   
1             1  Ovinos - Lana y Carne            1               Carne ovina   
2             1  Ovinos - Lana y Carne            1               Carne ovina   
3             1  Ovinos - Lana y Carne            1               Carne ovina   
4             1  Ovinos - Lana y Carne            1               Carne ovina   
...         ...                    ...          ...                       ...   
8050          1  Ovinos - Lana y Carne           18                     Ovino   
8051          1  Ovinos - Lana y Carne           18                     Ovino   
8052          1  Ovinos - Lana y Carne           18                     Ovino   
8053          1  Ovinos - Lana y Carne           18                     Ovino   
8054          1  Ovinos - Lana y Carne           18                     Ovino   

             indicador  unidad_de_medida  fuente frecuencia_nombre  \
0     Consumo aparente  tn res con hueso   MAGYP             Anual   
1     Consumo aparente  tn res con hueso   MAGYP             Anual   
2     Consumo aparente  tn res con hueso   MAGYP             Anual   
3     Consumo aparente  tn res con hueso   MAGYP             Anual   
4     Consumo aparente  tn res con hueso   MAGYP             Anual   
...                ...               ...     ...               ...   
8050    Stock al 31-03           cabezas  SENASA             Anual   
8051    Stock al 31-03           cabezas  SENASA             Anual   
8052    Stock al 31-03           cabezas  SENASA             Anual   
8053    Stock al 31-03           cabezas  SENASA             Anual   
8054    Stock al 31-03           cabezas  SENASA             Anual   

     cobertura_nombre  alcance_tipo  alcance_id alcance_nombre indice_tiempo  \
0            Nacional          PAIS         200      Argentina    1998-01-01   
1            Nacional          PAIS         200      Argentina    1999-01-01   
2            Nacional          PAIS         200      Argentina    2000-01-01   
3            Nacional          PAIS         200      Argentina    2001-01-01   
4            Nacional          PAIS         200      Argentina    2002-01-01   
...               ...           ...         ...            ...           ...   
8050         Nacional  DEPARTAMENTO       94015        USHUAIA    2015-01-01   
8051         Nacional  DEPARTAMENTO       94015        USHUAIA    2016-01-01   
8052         Nacional  DEPARTAMENTO       94015        USHUAIA    2018-01-01   
8053         Nacional  DEPARTAMENTO       94015        USHUAIA    2019-01-01   
8054         Nacional  DEPARTAMENTO       94015        USHUAIA    2020-01-01   

             valor  
0     39221.694655  
1     52247.903796  
2     67633.175045  
3     44560.747617  
4     44549.057091  
...            ...  
8050     81.000000  
8051     43.000000  
8052      4.000000  
8053      6.000000  
8054     27.000000  

[8055 rows x 14 columns]

Como se mencionó anteriormente se trata de un conjunto de datos un tanto mezclado, y será necesario analizarlo en detenimiento para encontrar los datos que necesitamos. Es un dataset de 8055 filas y 14 columnas, miremos las columnas que trae:

In [3]:
df.columns

Index(['sector_id', 'sector_nombre', 'variable_id',
       'actividad_producto_nombre', 'indicador', 'unidad_de_medida', 'fuente',
       'frecuencia_nombre', 'cobertura_nombre', 'alcance_tipo', 'alcance_id',
       'alcance_nombre', 'indice_tiempo', 'valor'],
      dtype='object')

La columna que parece decirnos el producto es "actividad_producto_nombre", por lo tanto veamos si en efecto es así:

In [4]:
df['actividad_producto_nombre'].unique()

array(['Carne ovina', 'Lana fina_ 20 micrones Rinde peine 55%',
       'Lana mediana_ 27 micrones Rinde peine 55%', 'Lana sucia',
       'Lana sucia cruza fina', 'Lana sucia fina', 'Lana sucia gruesa',
       'Lana sucia mediana', 'Ovino'], dtype=object)

Tenemos 9 "productos":
- Carne ovina
- Lana fina_ 20 micrones Rinde peine 55%
- Lana mediana_ 27 micrones Rinde peine 55%
- Lana sucia
- Lana sucia cruza fina
- Lana sucia fina
- Lana sucia gruesa
- Lana sucia mediana
- Ovino: Representa las existencias ovinas

Lo siguiente es buscar los productos que nos interesan pero que sean de la provincia del Chubut. Empezamos por la producción de carne.

In [5]:
resultado_carne = pd.DataFrame(columns=['actividad_producto_nombre','indicador', 'unidad_de_medida', 'frecuencia_nombre', 'alcance_tipo', 'alcance_nombre', 'indice_tiempo', 'valor'])
for i in df.index: 
    if "Carne ovina" == df['actividad_producto_nombre'][i]:
         resultado_carne = resultado_carne.append({
                                       'actividad_producto_nombre': df['actividad_producto_nombre'][i],
                                       'indicador': df['indicador'][i],
                                       'unidad_de_medida':df['unidad_de_medida'][i], 
                                       'frecuencia_nombre':df['frecuencia_nombre'][i],
                                       'alcance_tipo':df['alcance_tipo'][i],
                                       'alcance_nombre':df['alcance_nombre'][i],
                                       'indice_tiempo':df['indice_tiempo'][i],
                                       'valor':df['valor'][i],}, 
                                      ignore_index=True)
resultado_carne

actividad_producto_nombre         indicador  unidad_de_medida  \
0                Carne ovina  Consumo aparente  tn res con hueso   
1                Carne ovina  Consumo aparente  tn res con hueso   
2                Carne ovina  Consumo aparente  tn res con hueso   
3                Carne ovina  Consumo aparente  tn res con hueso   
4                Carne ovina  Consumo aparente  tn res con hueso   
..                       ...               ...               ...   
61               Carne ovina        Producción  tn res con hueso   
62               Carne ovina        Producción  tn res con hueso   
63               Carne ovina        Producción  tn res con hueso   
64               Carne ovina        Producción  tn res con hueso   
65               Carne ovina        Producción  tn res con hueso   

   frecuencia_nombre alcance_tipo alcance_nombre indice_tiempo         valor  
0              Anual         PAIS      Argentina    1998-01-01  39221.694655  
1              Anual         PAIS      Argentina    1999-01-01  52247.903796  
2              Anual         PAIS      Argentina    2000-01-01  67633.175045  
3              Anual         PAIS      Argentina    2001-01-01  44560.747617  
4              Anual         PAIS      Argentina    2002-01-01  44549.057091  
..               ...          ...            ...           ...           ...  
61             Anual         PAIS      Argentina    2015-01-01  57608.544032  
62             Anual         PAIS      Argentina    2016-01-01  51590.113369  
63             Anual         PAIS      Argentina    2017-01-01  50615.827729  
64             Anual         PAIS      Argentina    2018-01-01  49126.437112  
65             Anual         PAIS      Argentina    2019-01-01  53392.475808  

[66 rows x 8 columns]

Son 66 filas, entonces lo que hacemos es buscar si existen datos a nivel provincia o departamento

In [6]:
resultado_carne['alcance_tipo'].unique()

array(['PAIS'], dtype=object)

No hay datos especificados por provincia ni por departamento, por lo tanto este dataset no tiene información sobre la producción de carne de Chubut. Pasemos al siguiente producto que sería la lana fina de 20 micrones.

In [7]:
resultado_lana_20 = pd.DataFrame(columns=['actividad_producto_nombre','indicador', 'unidad_de_medida', 'frecuencia_nombre', 'alcance_tipo', 'alcance_nombre', 'indice_tiempo', 'valor'])
for i in df.index: 
    if "Lana fina_ 20 micrones Rinde peine 55%" == df['actividad_producto_nombre'][i]:
         resultado_lana_20 = resultado_lana_20.append({
                                       'actividad_producto_nombre': df['actividad_producto_nombre'][i],
                                       'indicador': df['indicador'][i],
                                       'unidad_de_medida':df['unidad_de_medida'][i], 
                                       'frecuencia_nombre':df['frecuencia_nombre'][i],
                                       'alcance_tipo':df['alcance_tipo'][i],
                                       'alcance_nombre':df['alcance_nombre'][i],
                                       'indice_tiempo':df['indice_tiempo'][i],
                                       'valor':df['valor'][i],
                                      }, 
                                      ignore_index=True)
resultado_lana_20

actividad_producto_nombre                     indicador  \
0    Lana fina_ 20 micrones Rinde peine 55%  Precio interno_de referencia   
1    Lana fina_ 20 micrones Rinde peine 55%  Precio interno_de referencia   
2    Lana fina_ 20 micrones Rinde peine 55%  Precio interno_de referencia   
3    Lana fina_ 20 micrones Rinde peine 55%  Precio interno_de referencia   
4    Lana fina_ 20 micrones Rinde peine 55%  Precio interno_de referencia   
..                                      ...                           ...   
210  Lana fina_ 20 micrones Rinde peine 55%  Precio interno_de referencia   
211  Lana fina_ 20 micrones Rinde peine 55%  Precio interno_de referencia   
212  Lana fina_ 20 micrones Rinde peine 55%  Precio interno_de referencia   
213  Lana fina_ 20 micrones Rinde peine 55%  Precio interno_de referencia   
214  Lana fina_ 20 micrones Rinde peine 55%  Precio interno_de referencia   

    unidad_de_medida frecuencia_nombre alcance_tipo alcance_nombre  \
0                US$           Mensual         PAIS      Argentina   
1                US$           Mensual         PAIS      Argentina   
2                US$           Mensual         PAIS      Argentina   
3                US$           Mensual         PAIS      Argentina   
4                US$           Mensual         PAIS      Argentina   
..               ...               ...          ...            ...   
210              US$           Mensual         PAIS      Argentina   
211              US$           Mensual         PAIS      Argentina   
212              US$           Mensual         PAIS      Argentina   
213              US$           Mensual         PAIS      Argentina   
214              US$           Mensual         PAIS      Argentina   

    indice_tiempo  valor  
0      2004-01-01   2.93  
1      2004-02-01   2.83  
2      2004-03-01   2.77  
3      2004-04-01   2.68  
4      2004-05-01   2.68  
..            ...    ...  
210    2021-07-01   4.43  
211    2021-08-01   4.01  
212    2021-09-01   3.87  
213    2021-10-01   4.08  
214    2021-11-01   3.99  

[215 rows x 8 columns]

Son 215 filas, entonces lo que hacemos es buscar si existen datos a nivel provincia o departamento

In [8]:
resultado_lana_20['alcance_tipo'].unique()

array(['PAIS'], dtype=object)

No hay datos especificados por provincia ni por departamento, por lo tanto este dataset no tiene información sobre la lana fina de 20 micrones de Chubut. Pasemos al siguiente producto que sería la lana mediana de 27 micrones.

In [9]:
resultado_lana_27 = pd.DataFrame(columns=['actividad_producto_nombre','indicador', 'unidad_de_medida', 'frecuencia_nombre', 'alcance_tipo', 'alcance_nombre', 'indice_tiempo', 'valor'])
for i in df.index:
    if "Lana mediana_ 27 micrones Rinde peine 55%" == df['actividad_producto_nombre'][i]:
         resultado_lana_27 = resultado_lana_27.append({
                                       'actividad_producto_nombre': df['actividad_producto_nombre'][i],
                                       'indicador': df['indicador'][i],
                                       'unidad_de_medida':df['unidad_de_medida'][i], 
                                       'frecuencia_nombre':df['frecuencia_nombre'][i],
                                       'alcance_tipo':df['alcance_tipo'][i],
                                       'alcance_nombre':df['alcance_nombre'][i],
                                       'indice_tiempo':df['indice_tiempo'][i],
                                       'valor':df['valor'][i],
                                         }, 
                                      ignore_index=True)
resultado_lana_27

actividad_producto_nombre                     indicador  \
0    Lana mediana_ 27 micrones Rinde peine 55%  Precio interno_de referencia   
1    Lana mediana_ 27 micrones Rinde peine 55%  Precio interno_de referencia   
2    Lana mediana_ 27 micrones Rinde peine 55%  Precio interno_de referencia   
3    Lana mediana_ 27 micrones Rinde peine 55%  Precio interno_de referencia   
4    Lana mediana_ 27 micrones Rinde peine 55%  Precio interno_de referencia   
..                                         ...                           ...   
210  Lana mediana_ 27 micrones Rinde peine 55%  Precio interno_de referencia   
211  Lana mediana_ 27 micrones Rinde peine 55%  Precio interno_de referencia   
212  Lana mediana_ 27 micrones Rinde peine 55%  Precio interno_de referencia   
213  Lana mediana_ 27 micrones Rinde peine 55%  Precio interno_de referencia   
214  Lana mediana_ 27 micrones Rinde peine 55%  Precio interno_de referencia   

    unidad_de_medida frecuencia_nombre alcance_tipo alcance_nombre  \
0                US$           Mensual         PAIS      Argentina   
1                US$           Mensual         PAIS      Argentina   
2                US$           Mensual         PAIS      Argentina   
3                US$           Mensual         PAIS      Argentina   
4                US$           Mensual         PAIS      Argentina   
..               ...               ...          ...            ...   
210              US$           Mensual         PAIS      Argentina   
211              US$           Mensual         PAIS      Argentina   
212              US$           Mensual         PAIS      Argentina   
213              US$           Mensual         PAIS      Argentina   
214              US$           Mensual         PAIS      Argentina   

    indice_tiempo  valor  
0      2004-01-01   1.90  
1      2004-02-01   1.91  
2      2004-03-01   1.87  
3      2004-04-01   1.80  
4      2004-05-01   1.78  
..            ...    ...  
210    2021-07-01   1.86  
211    2021-08-01   1.74  
212    2021-09-01   1.60  
213    2021-10-01   1.50  
214    2021-11-01   1.52  

[215 rows x 8 columns]

Son 215 filas, entonces lo que hacemos es buscar si existen datos a nivel provincia o departamento

In [10]:
resultado_lana_27['alcance_tipo'].unique()

array(['PAIS'], dtype=object)

No hay datos especificados por provincia ni por departamento, por lo tanto este dataset no tiene información sobre la lana mediana de 27 micrones de Chubut. Pasemos al siguiente producto que sería la lana sucia.

In [11]:
resultado_lana_sucia = pd.DataFrame(columns=['actividad_producto_nombre','indicador', 'unidad_de_medida', 'frecuencia_nombre', 'alcance_tipo', 'alcance_nombre', 'indice_tiempo', 'valor'])
for i in df.index: 
    if "Lana sucia" == df['actividad_producto_nombre'][i]:
         resultado_lana_sucia = resultado_lana_sucia.append({
                                       'actividad_producto_nombre': df['actividad_producto_nombre'][i],
                                       'indicador': df['indicador'][i],
                                       'unidad_de_medida':df['unidad_de_medida'][i], 
                                       'frecuencia_nombre':df['frecuencia_nombre'][i],
                                       'alcance_tipo':df['alcance_tipo'][i],
                                       'alcance_nombre':df['alcance_nombre'][i],
                                       'indice_tiempo':df['indice_tiempo'][i],
                                       'valor':df['valor'][i],
                                         }, 
                                      ignore_index=True)
resultado_lana_sucia

actividad_producto_nombre   indicador unidad_de_medida frecuencia_nombre  \
0                  Lana sucia  Producción               tn             Anual   
1                  Lana sucia  Producción               tn             Anual   
2                  Lana sucia  Producción               tn             Anual   
3                  Lana sucia  Producción               tn             Anual   
4                  Lana sucia  Producción               tn             Anual   
..                        ...         ...              ...               ...   
477                Lana sucia  Producción               tn             Anual   
478                Lana sucia  Producción               tn             Anual   
479                Lana sucia  Producción               tn             Anual   
480                Lana sucia  Producción               tn             Anual   
481                Lana sucia  Producción               tn             Anual   

    alcance_tipo alcance_nombre indice_tiempo      valor  
0          MUNDO          MUNDO    1990-01-01  3350481.0  
1          MUNDO          MUNDO    1991-01-01  2736390.0  
2          MUNDO          MUNDO    1992-01-01  2872360.0  
3          MUNDO          MUNDO    1993-01-01  2775281.0  
4          MUNDO          MUNDO    1994-01-01  2687770.0  
..           ...            ...           ...        ...  
477    PROVINCIA  INDETERMINADA    2003-01-01     5900.0  
478    PROVINCIA  INDETERMINADA    2004-01-01     7000.0  
479    PROVINCIA  INDETERMINADA    2005-01-01     6200.0  
480    PROVINCIA  INDETERMINADA    2006-01-01     4600.0  
481    PROVINCIA  INDETERMINADA    2007-01-01     3300.0  

[482 rows x 8 columns]

Son 482 filas, entonces lo que hacemos es buscar si existen datos a nivel provincia o departamento

In [12]:
resultado_lana_sucia['alcance_tipo'].unique()

array(['MUNDO', 'PAIS', 'PROVINCIA'], dtype=object)

Hay datos especificados por provincia, por lo tanto buscamos si hay de Chubut.

In [13]:
resultado_lana_sucia['alcance_nombre'].unique()

array(['MUNDO', 'Argentina', 'BUENOS AIRES', 'CATAMARCA', 'CORDOBA',
       'CORRIENTES', 'CHACO', 'CHUBUT', 'ENTRE RIOS', 'FORMOSA', 'JUJUY',
       'LA PAMPA', 'LA RIOJA', 'MENDOZA', 'MISIONES', 'NEUQUEN',
       'RIO NEGRO', 'SALTA', 'SAN JUAN', 'SAN LUIS', 'SANTA CRUZ',
       'SANTA FE', 'SANTIAGO DEL ESTERO', 'TUCUMAN', 'TIERRA DEL FUEGO',
       'INDETERMINADA'], dtype=object)

Efectivamente hay datos sobre la producción de lana sucia en Chubut, por lo tanto filtramos esos datos y los guardamos en un archivo csv.

In [14]:
resultado_lana_sucia_chubut = pd.DataFrame(columns=['actividad_producto_nombre','indicador', 'unidad_de_medida', 'frecuencia_nombre', 'alcance_tipo', 'alcance_nombre', 'indice_tiempo', 'valor'])
for i in resultado_lana_sucia.index: 
    if "CHUBUT" == resultado_lana_sucia['alcance_nombre'][i]:
         resultado_lana_sucia_chubut = resultado_lana_sucia_chubut.append({
                                       'actividad_producto_nombre': resultado_lana_sucia['actividad_producto_nombre'][i],
                                       'indicador': resultado_lana_sucia['indicador'][i],
                                       'unidad_de_medida':resultado_lana_sucia['unidad_de_medida'][i], 
                                       'frecuencia_nombre':resultado_lana_sucia['frecuencia_nombre'][i],
                                       'alcance_tipo':resultado_lana_sucia['alcance_tipo'][i],
                                       'alcance_nombre':resultado_lana_sucia['alcance_nombre'][i],
                                       'indice_tiempo':resultado_lana_sucia['indice_tiempo'][i],
                                       'valor':resultado_lana_sucia['valor'][i],}, 
                                      ignore_index=True)
resultado_lana_sucia_chubut

actividad_producto_nombre   indicador unidad_de_medida frecuencia_nombre  \
0                 Lana sucia  Producción               tn             Anual   
1                 Lana sucia  Producción               tn             Anual   
2                 Lana sucia  Producción               tn             Anual   
3                 Lana sucia  Producción               tn             Anual   
4                 Lana sucia  Producción               tn             Anual   
5                 Lana sucia  Producción               tn             Anual   
6                 Lana sucia  Producción               tn             Anual   
7                 Lana sucia  Producción               tn             Anual   
8                 Lana sucia  Producción               tn             Anual   
9                 Lana sucia  Producción               tn             Anual   
10                Lana sucia  Producción               tn             Anual   
11                Lana sucia  Producción               tn             Anual   
12                Lana sucia  Producción               tn             Anual   
13                Lana sucia  Producción               tn             Anual   
14                Lana sucia  Producción               tn             Anual   
15                Lana sucia  Producción               tn             Anual   
16                Lana sucia  Producción               tn             Anual   
17                Lana sucia  Producción               tn             Anual   
18                Lana sucia  Producción               tn             Anual   
19                Lana sucia  Producción               tn             Anual   
20                Lana sucia  Producción               tn             Anual   
21                Lana sucia  Producción               tn             Anual   
22                Lana sucia  Producción               tn             Anual   

   alcance_tipo alcance_nombre indice_tiempo     valor  
0     PROVINCIA         CHUBUT    2000-01-01  17000.00  
1     PROVINCIA         CHUBUT    2001-01-01  17000.00  
2     PROVINCIA         CHUBUT    2002-01-01  16900.00  
3     PROVINCIA         CHUBUT    2003-01-01  21300.00  
4     PROVINCIA         CHUBUT    2004-01-01  21900.00  
5     PROVINCIA         CHUBUT    2005-01-01  23200.00  
6     PROVINCIA         CHUBUT    2006-01-01  22000.00  
7     PROVINCIA         CHUBUT    2007-01-01  22000.00  
8     PROVINCIA         CHUBUT    2008-01-01  21600.00  
9     PROVINCIA         CHUBUT    2009-01-01  17120.00  
10    PROVINCIA         CHUBUT    2010-01-01  17120.00  
11    PROVINCIA         CHUBUT    2011-01-01  16890.00  
12    PROVINCIA         CHUBUT    2012-01-01  13900.00  
13    PROVINCIA         CHUBUT    2013-01-01  13280.00  
14    PROVINCIA         CHUBUT    2014-01-01  13900.00  
15    PROVINCIA         CHUBUT    2015-01-01  14400.00  
16    PROVINCIA         CHUBUT    2016-01-01  14000.00  
17    PROVINCIA         CHUBUT    2017-01-01  14000.00  
18    PROVINCIA         CHUBUT    2018-01-01  14000.00  
19    PROVINCIA         CHUBUT    2019-01-01  13473.00  
20    PROVINCIA         CHUBUT    2020-01-01  13500.00  
21    PROVINCIA         CHUBUT    2021-01-01  13339.00  
22    PROVINCIA         CHUBUT    2022-01-01  11551.34

In [15]:
resultado_lana_sucia_chubut.to_csv('datos_producidos/indicadores_ovinos/lana_sucia_chubut_2000-2022.csv', index=False)

Pasamos al siguiente producto que sería la lana sucia cruza fina.

In [16]:
resultado_lana_sucia_cruza_fina = pd.DataFrame(columns=['actividad_producto_nombre','indicador', 'unidad_de_medida', 'frecuencia_nombre', 'alcance_tipo', 'alcance_nombre', 'indice_tiempo', 'valor'])
for i in df.index: 
    if "Lana sucia cruza fina" == df['actividad_producto_nombre'][i]:
         resultado_lana_sucia_cruza_fina = resultado_lana_sucia_cruza_fina.append({
                                       'actividad_producto_nombre': df['actividad_producto_nombre'][i],
                                       'indicador': df['indicador'][i],
                                       'unidad_de_medida':df['unidad_de_medida'][i], 
                                       'frecuencia_nombre':df['frecuencia_nombre'][i],
                                       'alcance_tipo':df['alcance_tipo'][i],
                                       'alcance_nombre':df['alcance_nombre'][i],
                                       'indice_tiempo':df['indice_tiempo'][i],
                                       'valor':df['valor'][i],
                                         }, 
                                      ignore_index=True)
resultado_lana_sucia_cruza_fina

actividad_producto_nombre              indicador unidad_de_medida  \
0      Lana sucia cruza fina  Producción por finura               tn   
1      Lana sucia cruza fina  Producción por finura               tn   
2      Lana sucia cruza fina  Producción por finura               tn   
3      Lana sucia cruza fina  Producción por finura               tn   
4      Lana sucia cruza fina  Producción por finura               tn   
5      Lana sucia cruza fina  Producción por finura               tn   
6      Lana sucia cruza fina  Producción por finura               tn   
7      Lana sucia cruza fina  Producción por finura               tn   
8      Lana sucia cruza fina  Producción por finura               tn   
9      Lana sucia cruza fina  Producción por finura               tn   
10     Lana sucia cruza fina  Producción por finura               tn   
11     Lana sucia cruza fina  Producción por finura               tn   
12     Lana sucia cruza fina  Producción por finura               tn   
13     Lana sucia cruza fina  Producción por finura               tn   
14     Lana sucia cruza fina  Producción por finura               tn   
15     Lana sucia cruza fina  Producción por finura               tn   

   frecuencia_nombre alcance_tipo alcance_nombre indice_tiempo    valor  
0              Anual         PAIS      Argentina    1990-01-01  67250.0  
1              Anual         PAIS      Argentina    1991-01-01  58000.0  
2              Anual         PAIS      Argentina    1992-01-01  51100.0  
3              Anual         PAIS      Argentina    1993-01-01  42400.0  
4              Anual         PAIS      Argentina    1994-01-01  35000.0  
5              Anual         PAIS      Argentina    1995-01-01  31600.0  
6              Anual         PAIS      Argentina    1996-01-01  27400.0  
7              Anual         PAIS      Argentina    1997-01-01  25650.0  
8              Anual         PAIS      Argentina    1998-01-01  24100.0  
9              Anual         PAIS      Argentina    1999-01-01  25200.0  
10             Anual         PAIS      Argentina    2000-01-01  22000.0  
11             Anual         PAIS      Argentina    2001-01-01  21750.0  
12             Anual         PAIS      Argentina    2002-01-01  23400.0  
13             Anual         PAIS      Argentina    2003-01-01  22000.0  
14             Anual         PAIS      Argentina    2004-01-01  22000.0  
15             Anual         PAIS      Argentina    2005-01-01  22960.0

Son 15 filas por lo tanto ya se observa que solo tiene alcance a nivel pais. Pasamos al siguiente producto que seria la lana sucia fina.

In [17]:
resultado_lana_sucia_fina = pd.DataFrame(columns=['actividad_producto_nombre','indicador', 'unidad_de_medida', 'frecuencia_nombre', 'alcance_tipo', 'alcance_nombre', 'indice_tiempo', 'valor'])
for i in df.index: 
    if "Lana sucia fina" == df['actividad_producto_nombre'][i]:
         resultado_lana_sucia_fina = resultado_lana_sucia_fina.append({
                                       'actividad_producto_nombre': df['actividad_producto_nombre'][i],
                                       'indicador': df['indicador'][i],
                                       'unidad_de_medida':df['unidad_de_medida'][i], 
                                       'frecuencia_nombre':df['frecuencia_nombre'][i],
                                       'alcance_tipo':df['alcance_tipo'][i],
                                       'alcance_nombre':df['alcance_nombre'][i],
                                       'indice_tiempo':df['indice_tiempo'][i],
                                       'valor':df['valor'][i],
                                         }, 
                                      ignore_index=True)
resultado_lana_sucia_fina

actividad_producto_nombre              indicador unidad_de_medida  \
0             Lana sucia fina  Producción por finura               tn   
1             Lana sucia fina  Producción por finura               tn   
2             Lana sucia fina  Producción por finura               tn   
3             Lana sucia fina  Producción por finura               tn   
4             Lana sucia fina  Producción por finura               tn   
..                        ...                    ...              ...   
226           Lana sucia fina  Producción por finura               tn   
227           Lana sucia fina  Producción por finura               tn   
228           Lana sucia fina  Producción por finura               tn   
229           Lana sucia fina  Producción por finura               tn   
230           Lana sucia fina  Producción por finura               tn   

    frecuencia_nombre alcance_tipo    alcance_nombre indice_tiempo     valor  
0               Anual         PAIS         Argentina    1990-01-01  44150.00  
1               Anual         PAIS         Argentina    1991-01-01  40000.00  
2               Anual         PAIS         Argentina    1992-01-01  37200.00  
3               Anual         PAIS         Argentina    1993-01-01  41800.00  
4               Anual         PAIS         Argentina    1994-01-01  36000.00  
..                ...          ...               ...           ...       ...  
226             Anual    PROVINCIA  TIERRA DEL FUEGO    2018-01-01    500.00  
227             Anual    PROVINCIA  TIERRA DEL FUEGO    2019-01-01    448.00  
228             Anual    PROVINCIA  TIERRA DEL FUEGO    2020-01-01    448.00  
229             Anual    PROVINCIA  TIERRA DEL FUEGO    2021-01-01    602.00  
230             Anual    PROVINCIA  TIERRA DEL FUEGO    2022-01-01    350.29  

[231 rows x 8 columns]

Son 231 filas, entonces lo que hacemos es buscar si existen datos a nivel provincia o departamento

In [18]:
resultado_lana_sucia_fina['alcance_tipo'].unique()

array(['PAIS', 'PROVINCIA'], dtype=object)

Hay datos especificados por provincia, por lo tanto buscamos si hay de Chubut.

In [19]:
resultado_lana_sucia_fina['alcance_nombre'].unique()

array(['Argentina', 'BUENOS AIRES', 'CATAMARCA', 'CORDOBA', 'CORRIENTES',
       'CHACO', 'CHUBUT', 'ENTRE RIOS', 'FORMOSA', 'JUJUY', 'LA PAMPA',
       'LA RIOJA', 'MENDOZA', 'MISIONES', 'NEUQUEN', 'RIO NEGRO', 'SALTA',
       'SAN JUAN', 'SAN LUIS', 'SANTA CRUZ', 'SANTA FE',
       'SANTIAGO DEL ESTERO', 'TUCUMAN', 'TIERRA DEL FUEGO'], dtype=object)


Efectivamente hay datos sobre la producción de lana sucia fina en Chubut, por lo tanto filtramos esos datos y los guardamos en un archivo csv.

In [20]:
resultado_lana_sucia_fina_chubut = pd.DataFrame(columns=['actividad_producto_nombre','indicador', 'unidad_de_medida', 'frecuencia_nombre', 'alcance_tipo', 'alcance_nombre', 'indice_tiempo', 'valor'])
for i in resultado_lana_sucia_fina.index: 
    if "CHUBUT" == resultado_lana_sucia_fina['alcance_nombre'][i]:
         resultado_lana_sucia_fina_chubut = resultado_lana_sucia_fina_chubut.append({
                                       'actividad_producto_nombre': resultado_lana_sucia_fina['actividad_producto_nombre'][i],
                                       'indicador': resultado_lana_sucia_fina['indicador'][i],
                                       'unidad_de_medida':resultado_lana_sucia_fina['unidad_de_medida'][i], 
                                       'frecuencia_nombre':resultado_lana_sucia_fina['frecuencia_nombre'][i],
                                       'alcance_tipo':resultado_lana_sucia_fina['alcance_tipo'][i],
                                       'alcance_nombre':resultado_lana_sucia_fina['alcance_nombre'][i],
                                       'indice_tiempo':resultado_lana_sucia_fina['indice_tiempo'][i],
                                       'valor':resultado_lana_sucia_fina['valor'][i],}, 
                                      ignore_index=True)
resultado_lana_sucia_fina_chubut

actividad_producto_nombre              indicador unidad_de_medida  \
0           Lana sucia fina  Producción por finura               tn   
1           Lana sucia fina  Producción por finura               tn   
2           Lana sucia fina  Producción por finura               tn   
3           Lana sucia fina  Producción por finura               tn   
4           Lana sucia fina  Producción por finura               tn   
5           Lana sucia fina  Producción por finura               tn   
6           Lana sucia fina  Producción por finura               tn   
7           Lana sucia fina  Producción por finura               tn   
8           Lana sucia fina  Producción por finura               tn   

  frecuencia_nombre alcance_tipo alcance_nombre indice_tiempo     valor  
0             Anual    PROVINCIA         CHUBUT    2014-01-01  13500.00  
1             Anual    PROVINCIA         CHUBUT    2015-01-01  14200.00  
2             Anual    PROVINCIA         CHUBUT    2016-01-01  13700.00  
3             Anual    PROVINCIA         CHUBUT    2017-01-01  13700.00  
4             Anual    PROVINCIA         CHUBUT    2018-01-01  13700.00  
5             Anual    PROVINCIA         CHUBUT    2019-01-01  13230.00  
6             Anual    PROVINCIA         CHUBUT    2020-01-01  13230.00  
7             Anual    PROVINCIA         CHUBUT    2021-01-01  13072.00  
8             Anual    PROVINCIA         CHUBUT    2022-01-01  11320.31

In [21]:
resultado_lana_sucia_fina_chubut.to_csv('datos_producidos/indicadores_ovinos/lana_sucia_fina_chubut_2014-2022.csv', index=False)

Pasamos al siguiente producto que seria la lana sucia gruesa.

In [22]:
resultado_lana_sucia_gruesa = pd.DataFrame(columns=['actividad_producto_nombre','indicador', 'unidad_de_medida', 'frecuencia_nombre', 'alcance_tipo', 'alcance_nombre', 'indice_tiempo', 'valor'])
for i in df.index: 
    if "Lana sucia gruesa" == df['actividad_producto_nombre'][i]:
         resultado_lana_sucia_gruesa = resultado_lana_sucia_gruesa.append({
                                       'actividad_producto_nombre': df['actividad_producto_nombre'][i],
                                       'indicador': df['indicador'][i],
                                       'unidad_de_medida':df['unidad_de_medida'][i], 
                                       'frecuencia_nombre':df['frecuencia_nombre'][i],
                                       'alcance_tipo':df['alcance_tipo'][i],
                                       'alcance_nombre':df['alcance_nombre'][i],
                                       'indice_tiempo':df['indice_tiempo'][i],
                                       'valor':df['valor'][i],
                                         }, 
                                      ignore_index=True)
resultado_lana_sucia_gruesa

actividad_producto_nombre              indicador unidad_de_medida  \
0           Lana sucia gruesa  Producción por finura               tn   
1           Lana sucia gruesa  Producción por finura               tn   
2           Lana sucia gruesa  Producción por finura               tn   
3           Lana sucia gruesa  Producción por finura               tn   
4           Lana sucia gruesa  Producción por finura               tn   
..                        ...                    ...              ...   
226         Lana sucia gruesa  Producción por finura               tn   
227         Lana sucia gruesa  Producción por finura               tn   
228         Lana sucia gruesa  Producción por finura               tn   
229         Lana sucia gruesa  Producción por finura               tn   
230         Lana sucia gruesa  Producción por finura               tn   

    frecuencia_nombre alcance_tipo    alcance_nombre indice_tiempo    valor  
0               Anual         PAIS         Argentina    1990-01-01  10300.0  
1               Anual         PAIS         Argentina    1991-01-01  13000.0  
2               Anual         PAIS         Argentina    1992-01-01  10000.0  
3               Anual         PAIS         Argentina    1993-01-01   8300.0  
4               Anual         PAIS         Argentina    1994-01-01   9000.0  
..                ...          ...               ...           ...      ...  
226             Anual    PROVINCIA  TIERRA DEL FUEGO    2018-01-01      0.0  
227             Anual    PROVINCIA  TIERRA DEL FUEGO    2019-01-01      0.0  
228             Anual    PROVINCIA  TIERRA DEL FUEGO    2020-01-01      0.0  
229             Anual    PROVINCIA  TIERRA DEL FUEGO    2021-01-01      0.0  
230             Anual    PROVINCIA  TIERRA DEL FUEGO    2022-01-01      0.0  

[231 rows x 8 columns]

Son 231 filas, entonces lo que hacemos es buscar si existen datos a nivel provincia o departamento

In [23]:
resultado_lana_sucia_gruesa['alcance_tipo'].unique()

array(['PAIS', 'PROVINCIA'], dtype=object)

Hay datos especificados por provincia, por lo tanto buscamos si hay de Chubut.

In [24]:
resultado_lana_sucia_gruesa['alcance_nombre'].unique()

array(['Argentina', 'BUENOS AIRES', 'CATAMARCA', 'CORDOBA', 'CORRIENTES',
       'CHACO', 'CHUBUT', 'ENTRE RIOS', 'FORMOSA', 'JUJUY', 'LA PAMPA',
       'LA RIOJA', 'MENDOZA', 'MISIONES', 'NEUQUEN', 'RIO NEGRO', 'SALTA',
       'SAN JUAN', 'SAN LUIS', 'SANTA CRUZ', 'SANTA FE',
       'SANTIAGO DEL ESTERO', 'TUCUMAN', 'TIERRA DEL FUEGO'], dtype=object)

Efectivamente hay datos sobre la producción de lana sucia gruesa en Chubut, por lo tanto filtramos esos datos y los guardamos en un archivo csv.

In [25]:
resultado_lana_sucia_gruesa_chubut = pd.DataFrame(columns=['actividad_producto_nombre','indicador', 'unidad_de_medida', 'frecuencia_nombre', 'alcance_tipo', 'alcance_nombre', 'indice_tiempo', 'valor'])
for i in resultado_lana_sucia_gruesa.index: 
    if "CHUBUT" == resultado_lana_sucia_gruesa['alcance_nombre'][i]:
         resultado_lana_sucia_gruesa_chubut = resultado_lana_sucia_gruesa_chubut.append({
                                       'actividad_producto_nombre': resultado_lana_sucia_gruesa['actividad_producto_nombre'][i],
                                       'indicador': resultado_lana_sucia_gruesa['indicador'][i],
                                       'unidad_de_medida':resultado_lana_sucia_gruesa['unidad_de_medida'][i], 
                                       'frecuencia_nombre':resultado_lana_sucia_gruesa['frecuencia_nombre'][i],
                                       'alcance_tipo':resultado_lana_sucia_gruesa['alcance_tipo'][i],
                                       'alcance_nombre':resultado_lana_sucia_gruesa['alcance_nombre'][i],
                                       'indice_tiempo':resultado_lana_sucia_gruesa['indice_tiempo'][i],
                                       'valor':resultado_lana_sucia_gruesa['valor'][i],}, 
                                      ignore_index=True)
resultado_lana_sucia_gruesa_chubut

actividad_producto_nombre              indicador unidad_de_medida  \
0         Lana sucia gruesa  Producción por finura               tn   
1         Lana sucia gruesa  Producción por finura               tn   
2         Lana sucia gruesa  Producción por finura               tn   
3         Lana sucia gruesa  Producción por finura               tn   
4         Lana sucia gruesa  Producción por finura               tn   
5         Lana sucia gruesa  Producción por finura               tn   
6         Lana sucia gruesa  Producción por finura               tn   
7         Lana sucia gruesa  Producción por finura               tn   
8         Lana sucia gruesa  Producción por finura               tn   

  frecuencia_nombre alcance_tipo alcance_nombre indice_tiempo  valor  
0             Anual    PROVINCIA         CHUBUT    2014-01-01    0.0  
1             Anual    PROVINCIA         CHUBUT    2015-01-01    0.0  
2             Anual    PROVINCIA         CHUBUT    2016-01-01    0.0  
3             Anual    PROVINCIA         CHUBUT    2017-01-01    0.0  
4             Anual    PROVINCIA         CHUBUT    2018-01-01    0.0  
5             Anual    PROVINCIA         CHUBUT    2019-01-01    0.0  
6             Anual    PROVINCIA         CHUBUT    2020-01-01    0.0  
7             Anual    PROVINCIA         CHUBUT    2021-01-01    0.0  
8             Anual    PROVINCIA         CHUBUT    2022-01-01    0.0

In [26]:
resultado_lana_sucia_gruesa_chubut.to_csv('datos_producidos/indicadores_ovinos/lana_sucia_gruesa_chubut_2014-2022.csv', index=False)

Pasamos al siguiente producto que sería la lana sucia mediana.

In [27]:
resultado_lana_sucia_mediana = pd.DataFrame(columns=['actividad_producto_nombre','indicador', 'unidad_de_medida', 'frecuencia_nombre', 'alcance_tipo', 'alcance_nombre', 'indice_tiempo', 'valor'])
for i in df.index: 
    if "Lana sucia mediana" == df['actividad_producto_nombre'][i]:
         resultado_lana_sucia_mediana = resultado_lana_sucia_mediana.append({
                                       'actividad_producto_nombre': df['actividad_producto_nombre'][i],
                                       'indicador': df['indicador'][i],
                                       'unidad_de_medida':df['unidad_de_medida'][i], 
                                       'frecuencia_nombre':df['frecuencia_nombre'][i],
                                       'alcance_tipo':df['alcance_tipo'][i],
                                       'alcance_nombre':df['alcance_nombre'][i],
                                       'indice_tiempo':df['indice_tiempo'][i],
                                       'valor':df['valor'][i],
                                         }, 
                                      ignore_index=True)
resultado_lana_sucia_mediana

actividad_producto_nombre              indicador unidad_de_medida  \
0          Lana sucia mediana  Producción por finura               tn   
1          Lana sucia mediana  Producción por finura               tn   
2          Lana sucia mediana  Producción por finura               tn   
3          Lana sucia mediana  Producción por finura               tn   
4          Lana sucia mediana  Producción por finura               tn   
..                        ...                    ...              ...   
226        Lana sucia mediana  Producción por finura               tn   
227        Lana sucia mediana  Producción por finura               tn   
228        Lana sucia mediana  Producción por finura               tn   
229        Lana sucia mediana  Producción por finura               tn   
230        Lana sucia mediana  Producción por finura               tn   

    frecuencia_nombre alcance_tipo    alcance_nombre indice_tiempo     valor  
0               Anual         PAIS         Argentina    1990-01-01  16900.00  
1               Anual         PAIS         Argentina    1991-01-01  14000.00  
2               Anual         PAIS         Argentina    1992-01-01  11700.00  
3               Anual         PAIS         Argentina    1993-01-01  10500.00  
4               Anual         PAIS         Argentina    1994-01-01   8000.00  
..                ...          ...               ...           ...       ...  
226             Anual    PROVINCIA  TIERRA DEL FUEGO    2018-01-01   1100.00  
227             Anual    PROVINCIA  TIERRA DEL FUEGO    2019-01-01    857.00  
228             Anual    PROVINCIA  TIERRA DEL FUEGO    2020-01-01    952.00  
229             Anual    PROVINCIA  TIERRA DEL FUEGO    2021-01-01    492.00  
230             Anual    PROVINCIA  TIERRA DEL FUEGO    2022-01-01    744.37  

[231 rows x 8 columns]

Son 231 filas, entonces lo que hacemos es buscar si existen datos a nivel provincia o departamento

In [28]:
resultado_lana_sucia_mediana['alcance_tipo'].unique()

array(['PAIS', 'PROVINCIA'], dtype=object)

Hay datos especificados por provincia, por lo tanto buscamos si hay de Chubut.

In [29]:
resultado_lana_sucia_mediana['alcance_nombre'].unique()

array(['Argentina', 'BUENOS AIRES', 'CATAMARCA', 'CORDOBA', 'CORRIENTES',
       'CHACO', 'CHUBUT', 'ENTRE RIOS', 'FORMOSA', 'JUJUY', 'LA PAMPA',
       'LA RIOJA', 'MENDOZA', 'MISIONES', 'NEUQUEN', 'RIO NEGRO', 'SALTA',
       'SAN JUAN', 'SAN LUIS', 'SANTA CRUZ', 'SANTA FE',
       'SANTIAGO DEL ESTERO', 'TUCUMAN', 'TIERRA DEL FUEGO'], dtype=object)

Efectivamente hay datos sobre la producción de lana sucia mediana en Chubut, por lo tanto filtramos esos datos y los guardamos en un archivo csv.

In [30]:
resultado_lana_sucia_mediana_chubut = pd.DataFrame(columns=['actividad_producto_nombre','indicador', 'unidad_de_medida', 'frecuencia_nombre', 'alcance_tipo', 'alcance_nombre', 'indice_tiempo', 'valor'])
for i in resultado_lana_sucia_gruesa.index: 
    if "CHUBUT" == resultado_lana_sucia_gruesa['alcance_nombre'][i]:
         resultado_lana_sucia_mediana_chubut = resultado_lana_sucia_mediana_chubut.append({
                                       'actividad_producto_nombre': resultado_lana_sucia_mediana['actividad_producto_nombre'][i],
                                       'indicador': resultado_lana_sucia_mediana['indicador'][i],
                                       'unidad_de_medida':resultado_lana_sucia_mediana['unidad_de_medida'][i], 
                                       'frecuencia_nombre':resultado_lana_sucia_mediana['frecuencia_nombre'][i],
                                       'alcance_tipo':resultado_lana_sucia_mediana['alcance_tipo'][i],
                                       'alcance_nombre':resultado_lana_sucia_mediana['alcance_nombre'][i],
                                       'indice_tiempo':resultado_lana_sucia_mediana['indice_tiempo'][i],
                                       'valor':resultado_lana_sucia_mediana['valor'][i],}, 
                                      ignore_index=True)
resultado_lana_sucia_mediana_chubut

actividad_producto_nombre              indicador unidad_de_medida  \
0        Lana sucia mediana  Producción por finura               tn   
1        Lana sucia mediana  Producción por finura               tn   
2        Lana sucia mediana  Producción por finura               tn   
3        Lana sucia mediana  Producción por finura               tn   
4        Lana sucia mediana  Producción por finura               tn   
5        Lana sucia mediana  Producción por finura               tn   
6        Lana sucia mediana  Producción por finura               tn   
7        Lana sucia mediana  Producción por finura               tn   
8        Lana sucia mediana  Producción por finura               tn   

  frecuencia_nombre alcance_tipo alcance_nombre indice_tiempo   valor  
0             Anual    PROVINCIA         CHUBUT    2014-01-01  400.00  
1             Anual    PROVINCIA         CHUBUT    2015-01-01  200.00  
2             Anual    PROVINCIA         CHUBUT    2016-01-01  300.00  
3             Anual    PROVINCIA         CHUBUT    2017-01-01  300.00  
4             Anual    PROVINCIA         CHUBUT    2018-01-01  300.00  
5             Anual    PROVINCIA         CHUBUT    2019-01-01  243.00  
6             Anual    PROVINCIA         CHUBUT    2020-01-01  270.00  
7             Anual    PROVINCIA         CHUBUT    2021-01-01  267.00  
8             Anual    PROVINCIA         CHUBUT    2022-01-01  231.03

In [31]:
resultado_lana_sucia_mediana_chubut.to_csv('datos_producidos/indicadores_ovinos/lana_sucia_mediana_chubut_2014-2022.csv', index=False)

Pasamos al siguiente producto que sería la existencia de ovinos.

In [32]:
resultado_ovinos = pd.DataFrame(columns=['actividad_producto_nombre','indicador', 'unidad_de_medida', 'frecuencia_nombre', 'alcance_tipo', 'alcance_nombre', 'indice_tiempo', 'valor'])
for i in df.index: 
    if "Ovino" == df['actividad_producto_nombre'][i]:
         resultado_ovinos = resultado_ovinos.append({
                                       'actividad_producto_nombre': df['actividad_producto_nombre'][i],
                                       'indicador': df['indicador'][i],
                                       'unidad_de_medida':df['unidad_de_medida'][i], 
                                       'frecuencia_nombre':df['frecuencia_nombre'][i],
                                       'alcance_tipo':df['alcance_tipo'][i],
                                       'alcance_nombre':df['alcance_nombre'][i],
                                       'indice_tiempo':df['indice_tiempo'][i],
                                       'valor':df['valor'][i],
                                         }, 
                                      ignore_index=True)
resultado_ovinos

actividad_producto_nombre       indicador unidad_de_medida  \
0                        Ovino  Faena estimada          cabezas   
1                        Ovino  Faena estimada          cabezas   
2                        Ovino  Faena estimada          cabezas   
3                        Ovino  Faena estimada          cabezas   
4                        Ovino  Faena estimada          cabezas   
...                        ...             ...              ...   
6363                     Ovino  Stock al 31-03          cabezas   
6364                     Ovino  Stock al 31-03          cabezas   
6365                     Ovino  Stock al 31-03          cabezas   
6366                     Ovino  Stock al 31-03          cabezas   
6367                     Ovino  Stock al 31-03          cabezas   

     frecuencia_nombre  alcance_tipo alcance_nombre indice_tiempo  \
0                Anual          PAIS      Argentina    1998-01-01   
1                Anual          PAIS      Argentina    1999-01-01   
2                Anual          PAIS      Argentina    2000-01-01   
3                Anual          PAIS      Argentina    2001-01-01   
4                Anual          PAIS      Argentina    2002-01-01   
...                ...           ...            ...           ...   
6363             Anual  DEPARTAMENTO        USHUAIA    2015-01-01   
6364             Anual  DEPARTAMENTO        USHUAIA    2016-01-01   
6365             Anual  DEPARTAMENTO        USHUAIA    2018-01-01   
6366             Anual  DEPARTAMENTO        USHUAIA    2019-01-01   
6367             Anual  DEPARTAMENTO        USHUAIA    2020-01-01   

             valor  
0     1.993746e+06  
1     2.997181e+06  
2     3.653250e+06  
3     2.411242e+06  
4     2.535396e+06  
...            ...  
6363  8.100000e+01  
6364  4.300000e+01  
6365  4.000000e+00  
6366  6.000000e+00  
6367  2.700000e+01  

[6368 rows x 8 columns]

Son 6368 filas, entonces lo que hacemos es buscar si existen datos a nivel provincia o departamento

In [33]:
resultado_ovinos['alcance_tipo'].unique()

array(['PAIS', 'DEPARTAMENTO'], dtype=object)

Hay datos especificados por departamento, por lo tanto buscamos si estan los de Chubut.

In [34]:
departamentos = resultado_ovinos['alcance_nombre'].unique().tolist()
departamentos_chubut = ["BIEDMA", "CUSHAMEN", "ESCALANTE","FUTALEUFU", "GAIMAN", "GASTRE","LANGUIÑEO", "MARTIRES", "PASO DE INDIOS", "RAWSON", "RIO SENGUER","SARMIENTO", "TEHUELCHES", "TELSEN"]

def compare(lst1, lst2, ini=0):
    n = len(lst2)
    return lst1[ini:ini+n] == lst2

def indexes(lst, x):
    return (i for (i,y) in enumerate(lst) if x==y)

any(compare(departamentos, departamentos_chubut, i) for i in indexes(departamentos, departamentos_chubut[0]))

True

Segun la función anterior están los departamentos de Chubut por lo tanto hay datos sobre las existencias ovinas en la provincia. Lo siguiente es filtrar esos datos y guardalos en un archivo csv.

In [35]:
departamentos_chubut = ["BIEDMA", "CUSHAMEN", "ESCALANTE","FUTALEUFU", "GAIMAN", "GASTRE","LANGUIÑEO", "MARTIRES", "PASO DE INDIOS", "RAWSON", "RIO SENGUER","SARMIENTO", "TEHUELCHES", "TELSEN"]

resultado_ovinos_chubut = pd.DataFrame(columns=['actividad_producto_nombre','indicador', 'unidad_de_medida', 'frecuencia_nombre', 'alcance_tipo', 'alcance_nombre', 'indice_tiempo', 'valor'])
for i in resultado_ovinos.index: 
    if resultado_ovinos['alcance_nombre'][i] in departamentos_chubut:
         resultado_ovinos_chubut = resultado_ovinos_chubut.append({
                                       'actividad_producto_nombre': resultado_ovinos['actividad_producto_nombre'][i],
                                       'indicador': resultado_ovinos['indicador'][i],
                                       'unidad_de_medida':resultado_ovinos['unidad_de_medida'][i], 
                                       'frecuencia_nombre':resultado_ovinos['frecuencia_nombre'][i],
                                       'alcance_tipo':resultado_ovinos['alcance_tipo'][i],
                                       'alcance_nombre':resultado_ovinos['alcance_nombre'][i],
                                       'indice_tiempo':resultado_ovinos['indice_tiempo'][i],
                                       'valor':resultado_ovinos['valor'][i],}, 
                                      ignore_index=True)
resultado_ovinos_chubut

actividad_producto_nombre       indicador unidad_de_medida  \
0                       Ovino  Stock al 31-03          cabezas   
1                       Ovino  Stock al 31-03          cabezas   
2                       Ovino  Stock al 31-03          cabezas   
3                       Ovino  Stock al 31-03          cabezas   
4                       Ovino  Stock al 31-03          cabezas   
..                        ...             ...              ...   
216                     Ovino  Stock al 31-03          cabezas   
217                     Ovino  Stock al 31-03          cabezas   
218                     Ovino  Stock al 31-03          cabezas   
219                     Ovino  Stock al 31-03          cabezas   
220                     Ovino  Stock al 31-03          cabezas   

    frecuencia_nombre  alcance_tipo alcance_nombre indice_tiempo     valor  
0               Anual  DEPARTAMENTO         BIEDMA    2008-01-01  291332.0  
1               Anual  DEPARTAMENTO         BIEDMA    2009-01-01  271568.0  
2               Anual  DEPARTAMENTO         BIEDMA    2010-01-01  241263.0  
3               Anual  DEPARTAMENTO         BIEDMA    2011-01-01  216823.0  
4               Anual  DEPARTAMENTO         BIEDMA    2012-01-01  193849.0  
..                ...           ...            ...           ...       ...  
216             Anual  DEPARTAMENTO      SARMIENTO    2016-01-01     910.0  
217             Anual  DEPARTAMENTO      SARMIENTO    2017-01-01    1083.0  
218             Anual  DEPARTAMENTO      SARMIENTO    2018-01-01    1106.0  
219             Anual  DEPARTAMENTO      SARMIENTO    2019-01-01    1299.0  
220             Anual  DEPARTAMENTO      SARMIENTO    2020-01-01    1300.0  

[221 rows x 8 columns]

Son 221 filas, por lo tanto verificamos que esten todos los departamentos de Chubut.

In [36]:
resultado_ovinos_chubut['alcance_nombre'].unique()

array(['BIEDMA', 'CUSHAMEN', 'ESCALANTE', 'FUTALEUFU', 'GAIMAN', 'GASTRE',
       'LANGUIÑEO', 'MARTIRES', 'PASO DE INDIOS', 'RAWSON', 'RIO SENGUER',
       'SARMIENTO', 'TEHUELCHES', 'TELSEN'], dtype=object)

Una vez comprobado que los departamentos estan, los guardamos. 

In [37]:
resultado_ovinos_chubut.to_csv('datos_producidos/indicadores_ovinos/ovinos_chubut_2008-2020.csv', index=False)